### PR TITLE
Cog refactor

### PR DESCRIPTION
**Pull request by Andrewki44**
Cogs reviewed and commands converted to slash commands.

`Admin` & `Trueskill` cogs added

To Do:
- Move as much out of `commands` as possible, into organised groups & cogs

### CODE REVIEW

**Comment by Andrewki44** [comment]
`common`, `list` & `random` cogs added.
Some reorganization of commands to `list.py`.

**Comment by Andrewki44** [comment]
I believe this is now in the final state for now.
Some of the most common / complicated commands have not been moved yet (`!add`, `!del`, `!sub` & `!autosub`) due to concerns over variable arguments in interactions, and not wanting to hold up everything else in the meantime.

I've tested fairly well as I've gone, and will do some more too.

I have maintained functionality in commands as-is as much as possible, but have made some minor changes necessary to convert to app_commands, as well as implementing some new checks, reorganizing cogs & utils, etc.

Appreciate any & all feedback

File: discord_bots/cogs/common.py [hunk 1]
```diff
@@ -0,0 +1,574 @@
...
+                if current in name:
+                    choices.append(
+                        app_commands.Choice(
+                            name=name,
+                            value=name,
+                        )
+                    )
+        return choices
+
+    async def queue_autocomplete(self, interaction: Interaction, current: str):
```
Comment (mcoot) [inline discord_bots/cogs/common.py:561]: I see some of these e.g. queue_autocomplete are defined identically in multiple cogs (this one doesn't appear to have any references). Could these be combined to a central place?

**Review by mcoot — COMMENTED** [review]
Broadly looks good to me. I can't claim to have validated the logic of every cog, but I looked at the first couple and the structure makes sense. A few small questions, but broadly speaking I think it's good to go. Would be good to get at least an additional reviewer before merge.

File: discord_bots/cogs/admin.py [hunk 2]
```diff
@@ -0,0 +1,692 @@
...
+        await interaction.response.send_message(
+            embed=Embed(description=f"Command `{name}` added", colour=Colour.green())
+        )
+
+    @group.command(
+        name="createdbbackup", description="Creates a backup of the database"
+    )
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    async def createdbbackup(self, interaction: Interaction):
```
[the remark this answers is not in the document]
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:248]: It is, but I converted the command over as-is.
Outside of the scope of this PR.

I did take a quick look at what was involved with the dockerized pgsql, but was more than I was willing to dive into as a part of this.

File: discord_bots/cogs/admin.py [hunk 3]
```diff
@@ -0,0 +1,692 @@
...
+                + finished_game_str(game),
+                colour=Colour.green(),
+            )
+        )
+
+    @group.command(name="remove", description="Remove an admin")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(member="Player to be removed as admin")
+    async def removeadmin(self, interaction: Interaction, member: Member):
```
[the remark this answers is not in the document]
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:445]: For the grouping, there was several ways it could've been done, but I decided on just having it be alphabetical in most circumstances. 
I think `utils.py` might be the only exception to that.

Further subdivision / grouping can certainly be looked at down the road, but as it is, none of these cogs exceed the 25 command maximum.

File: discord_bots/cogs/common.py (hunk 1 above)
Comment (Andrewki44) [reply discord_bots/cogs/common.py:561]: Good catch, removed the unused autocomplete function from `common.py`.

You're correct that these are replicated a couple times, and 95% of them are identical. 
My only concern with externalising them from each individual cog was the decorator requirement.
As the autocomplete function is decorated with the slash command function (and so the slash command must be in scope), and not the other way around, it seemed tricky to move them elsewhere.

Creating a new `AutocompleteCommands` cog (or similar) doesn't compile, as you cannot use the function from another cog as the decorator inside the new cog.
Similarly, creating a new file (that is not a cog), and just loading the defs into there (similar to `commands` or `utils` compiles, but does not resolve the results when running the slash command it is attempting to autocomplete.

TLDR;
The autocomplete function seems to need to be in the same scope as the function it is referring to.

Comment (Andrewki44) [reply discord_bots/cogs/common.py]: The specific def referenced is removed in d4871a3

File: discord_bots/checks.py [hunk 4]
```diff
@@ -90,23 +108,50 @@ async def is_admin_app_command(interaction: Interaction) -> bool:
...
-
-    if not interaction:
-        return False
-
-    if not ECONOMY_ENABLED:
-        await interaction.response.send_message(
-            "Player economy is disabled",
-            ephemeral=True
-        )
+    if not interaction.channel:
```
[the remark this answers is not in the document]
Comment (Andrewki44) [reply discord_bots/checks.py:115]: Separating the conditional clauses removed a typing error that was present.
As if you try to interaction with `interaction.channel` before determining that it is `not None`, it presents an error.

Having a code path that returns out to False before checking if the `interaction.channel.id == CHANNEL_ID` removes the error from presenting.

Realistically, yes it could likely be compressed a bit, but I figured going for syntactic accuracy was preferable.

[the remark this answers is not in the document]
Comment (mcoot) [reply discord_bots/checks.py:115]: Maybe the typechecker can't narrow it but the `not interaction.channel` being first in the condition should resolve that issue since it short-circuits?

[the remark this answers is not in the document]
Comment (Andrewki44) [reply discord_bots/checks.py]: You make a fair point.
I've simplified the check in [acb7a20](https://github.com/tribesonecommunity/discord-bots/pull/93/commits/acb7a204e4d05402c361bc73c41b4638d6e35efa), and tested it as successful.

File: discord_bots/cogs/admin.py [hunk 5]
```diff
@@ -0,0 +1,692 @@
...
+
+    @group.command(name="ban", description="Bans player from queues")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(member="Player to be banned")
+    async def ban(self, interaction: Interaction, member: Member):
+        """TODO: remove player from queues"""
+        session: SQLAlchemySession
+        with Session() as session:
+            players = session.query(Player).filter(Player.id == member.id).all()
```
Comment (StephenBartos) [inline discord_bots/cogs/admin.py:139]: `.first()` can definitely be used here instead to simplify the logic, but since it was like this before you can just add a TODO. Up to you if you want to refactor it or not, because the logic that follows assumes the correct player is `players[0]`

File: discord_bots/cogs/admin.py [hunk 6]
```diff
@@ -0,0 +1,692 @@
...
+                        embed=Embed(
+                            description="Server configured successfully!",
+                            colour=Colour.green(),
+                        ),
+                        ephemeral=True,
+                    )
+        else:
+            await interaction.response.send_message(
+                embed=Embed(
+                    description="Command must be run from within a guild",
```
Comment (StephenBartos) [inline discord_bots/cogs/admin.py:210]: the `app_commands.guild_only` decorator can get rid of this check, if you want to simplify it a bit. Just `assert guild` at the top if you do that

File: discord_bots/cogs/admin.py [hunk 7]
```diff
@@ -0,0 +1,692 @@
...
+                colour=Colour.green(),
+            )
+        )
+
+    @group.command(
+        name="editgamewinner", description="Edit the winner of a finished game"
+    )
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(game_id="Finished game id", outcome="<tie|be|ds>")
```
Comment (StephenBartos) [inline discord_bots/cogs/admin.py:391]: does `outcome` need to be a string too or can it also be a list of `Literals`?

File: discord_bots/cogs/admin.py [hunk 8]
```diff
@@ -0,0 +1,692 @@
...
+    async def editgamewinner(
+        self,
+        interaction: Interaction,
+        game_id: str,
+        outcome: Literal["Tie", "BE", "DS"],
```
Comment (StephenBartos) [inline discord_bots/cogs/admin.py:396]: For the future, we may want to move away from the BE/DS convention here, and just do `team0, team1` or `team1, team2`.

File: discord_bots/cogs/admin.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (StephenBartos) [reply discord_bots/cogs/admin.py:445]: As we've discussed over discord, further command grouping can come down with more general refactors to the _logic_ of the commands. I don't really care about the ordering within the files, since the darker doesn't do it for you and you can always use an IDE or grep to find whatever one you want. Of course that would be nice to have, but unless there's some automated way to do it via the pre-commit hook, then I don't want anyone to have to do that themselves by hand. Alphabetical is preferred

File: discord_bots/cogs/admin.py [hunk 9]
```diff
@@ -0,0 +1,692 @@
...
+        )
+
+    @group.command(name="remove", description="Remove an admin")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(member="Player to be removed as admin")
+    async def removeadmin(self, interaction: Interaction, member: Member):
+        session: SQLAlchemySession
+        with Session() as session:
+            players = session.query(Player).filter(Player.id == member.id).all()
```
Comment (StephenBartos) [inline discord_bots/cogs/admin.py:448]: same thing as before, where `.first()` can be used

File: discord_bots/cogs/admin.py [hunk 10]
```diff
@@ -0,0 +1,692 @@
...
+                colour=Colour.green(),
+            )
+        )
+
+    @group.command(name="removerole", description="Remove an admin role")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(role="Role to be removed as admin")
+    async def removeadminrole(self, interaction: Interaction, role: Role):
+        if interaction.guild:
```
Comment (StephenBartos) [inline discord_bots/cogs/admin.py:472]: Use `app_commands.guild_only` to simplify

File: discord_bots/cogs/admin.py [hunk 11]
```diff
@@ -0,0 +1,692 @@
...
+                    description=f"Command `{name}` removed",
+                    colour=Colour.green(),
+                )
+            )
+
+    @group.command(name="removedbbackup", description="Remove a database backup")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(db_filename="Name of backup file")
+    async def removedbbackup(self, interaction: Interaction, db_filename: str):
```
Comment (StephenBartos) [inline discord_bots/cogs/admin.py:542]: the db specific commands don't work with postgres anymore, so it should be marked clearly both with a comment and an early return inside of the function that tells the user so.

File: discord_bots/cogs/admin.py [hunk 12]
```diff
@@ -0,0 +1,692 @@
...
+                embed=Embed(
+                    description=f"DB backup {db_filename} removed",
+                    colour=Colour.green(),
+                )
+            )
+
+    @group.command(name="restart", description="Restart the bot")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    async def restart(self, interaction: Interaction):
```
Comment (StephenBartos) [inline discord_bots/cogs/admin.py:577]: This might be dangerous to keep around with the dockerized setup, since any admin can use it as well. I'm not sure how it behaves with docker, though

File: discord_bots/cogs/admin.py [hunk 13]
```diff
@@ -0,0 +1,692 @@
...
+                colour=Colour.blue(),
+            )
+        )
+        os.execv(sys.executable, ["python", "-m", "discord_bots.main"])
+
+    @group.command(name="setbias", description="Set team bias")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(member="Member to bias", amount="Bias value")
+    async def setbias(self, interaction: Interaction, member: Member, amount: float):
```
Comment (StephenBartos) [inline discord_bots/cogs/admin.py:590]: Since these are "meme" functions, we may just want to leave them as prefix command or (if there's a way to) hide them. Not really sure, since they just pollute the command list

File: discord_bots/cogs/admin.py [hunk 14]
```diff
@@ -0,0 +1,692 @@
...
+        )
+
+    @group.command(name="unban", description="Unban player")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(member="Player to be unbanned")
+    async def unban(self, interaction: Interaction, member: Member):
+        session: SQLAlchemySession
+        with Session() as session:
+            players = session.query(Player).filter(Player.id == member.id).all()
```
Comment (StephenBartos) [inline discord_bots/cogs/admin.py:654]: same as before: use `.first()`

**Review by StephenBartos — COMMENTED** [review]
Reviewed checks.py and admin.py:

I understand that you migrated these commands over as-is, so I don't expect you to address every single comment with a code change; for nearly all of these, just adding a `TODO` or some other comment is good enough for a second pass in the future. I'm just taking this chance to review each of these commands. Awesome work so far!

- [x] checks.py
- [x] admin.py
- [x] base.py
- [ ] category.py
- [ ] common.py
- [ ] economy.py
- [ ] in_progress_game.py
- [ ] list.py
- [ ] map.py
- [ ] notification.py
- [ ] player.py
- [ ] queue.py
- [ ] raffle.py
- [ ] random.py
- [ ] rotation.py
- [ ] schedule.py
- [ ] trueskill.py
- [ ] vote.py
- [ ] commands.py
- [ ] config.py
- [ ] main.py
- [ ] models.py
- [ ] utils.py

File: discord_bots/checks.py [hunk 15]
```diff
@@ -6,10 +6,28 @@
...
+    """
+
+    if not interaction:
+        return False
+
+    if not ECONOMY_ENABLED:
+        await interaction.response.send_message(
+            "Player economy is disabled",
+            ephemeral=True
+        )
```
[the remark this answers is not in the document]
Comment (Andrewki44) [reply discord_bots/checks.py:25]: Updated in next commit

File: discord_bots/cogs/admin.py [hunk 16]
```diff
@@ -0,0 +1,692 @@
...
+                        )
+                    )
+
+    @group.command(name="addrole", description="Add an admin role")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(role="Role to be made admin")
+    async def addadminrole(self, interaction: Interaction, role: Role):
+        if interaction.guild:
+            if not role in interaction.guild.roles:
```
[the remark this answers is not in the document]
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:97]: Updated in next commit

File: discord_bots/cogs/admin.py [hunk 17]
```diff
@@ -0,0 +1,692 @@
...
+            with Session() as session:
+                admin_role: AdminRole | None = (
+                    session.query(AdminRole)
+                    .filter(AdminRole.role_id == role.id)
+                    .first()
+                )
+                if admin_role:
+                    await interaction.response.send_message(
+                        embed=Embed(
+                            description="Role is already an admin", colour=Colour.blue()
```
[the remark this answers is not in the document]
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:117]: Updated in next commit

File: discord_bots/cogs/admin.py (hunk 5 above)
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:139]: Replaced `players` with `player: Player | None`, and modified function logic to understand the singular nature of `.first()` as opposed to `.all()`

File: discord_bots/cogs/admin.py (hunk 6 above)
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:210]: Added `@app_commands.guild_only()`, and simplified function logic to remove the `if interaction.guild`

File: discord_bots/cogs/admin.py (hunk 7 above)
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:391]: Not sure if I'm misunderstanding, but `outcome` already is a Literal list of of strings.
`outcome: Literal["Tie", "BE", "DS"]`

File: discord_bots/cogs/admin.py (hunk 8 above)
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:396]: Added a `TODO`

File: discord_bots/cogs/admin.py (hunk 7 above)
Comment (StephenBartos) [reply discord_bots/cogs/admin.py:391]: I think I misunderstood myself, since this is just for the description. I don't think outcome even needs a description, since it will be autofilled to 3 valid options: "Tie", "BE", or "DS" -- my point was that this doesn't match precisely

Comment (Andrewki44) [reply discord_bots/cogs/admin.py:391]: Understood.
I do prefer to have the `.describe()` there for it anyway, just for completeness sake. I have updated it thought to match the `Literal` (`"<Tie|BE|DS>"`)

File: discord_bots/cogs/admin.py (hunk 9 above)
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:448]: Changed to `.first()`, and subsequent logic updated.

File: discord_bots/cogs/admin.py (hunk 7 above)
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:391]: Updated description to `"Tie, BE, DS"` as discussed in Discord

File: discord_bots/cogs/admin.py (hunk 10 above)
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:472]: added `@app_commands.guild_only()`, and modified subsequent logic to match.

File: discord_bots/cogs/admin.py (hunk 11 above)
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:542]: Added check on `config.DATABASE_URI` to return early if setup for postgres.
Added comments & `TODO`.

Similarly updated `createdbbackup()`

File: discord_bots/cogs/admin.py (hunk 12 above)
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:577]: The early return on postgres db's will be sufficient for now to block any bad interactions. In future this & `createdbbackup()` should be updated to support the dockerized db.
`TODO` added

File: discord_bots/cogs/admin.py (hunk 13 above)
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:590]: I would rather not leave anything as prefix / context commands, as the direction from discord itself is to move away from context commands entirely.
https://support-dev.discord.com/hc/en-us/articles/4404772028055-Message-Content-Privileged-Intent-FAQ

File: discord_bots/cogs/admin.py (hunk 14 above)
Comment (Andrewki44) [reply discord_bots/cogs/admin.py:654]: Updated to use `.first()`, along with the subsequent logic.

**Comment by Andrewki44** [comment]
`player.leaderboard_enabled` was not being considered in `print_leaderboard()`.
Fixed in ffbb2f5

File: discord_bots/cogs/admin.py (hunk 13 above)
Comment (StephenBartos) [reply discord_bots/cogs/admin.py:615]: If the prefix command @mentions the bot or is used inside of the bot's DM, then it doesn't required the message content privileged intent.

File: discord_bots/cogs/category.py [hunk 18]
```diff
@@ -0,0 +1,337 @@
...
+        await interaction.response.send_message(
+            embed=Embed(
+                description=f"The minimum number of games required to appear on the leaderboard for category **{category.name}** is now **{min_num_games}**",
+                colour=Colour.green(),
+            )
+        )
+
+    @group.command(name="show", description="Show category details")
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(category_name="Existing category")
```
Comment (StephenBartos) [inline discord_bots/cogs/category.py:276]: `app_copmmands.rename` `category_name` to `category`

File: discord_bots/cogs/common.py [hunk 19]
```diff
@@ -0,0 +1,559 @@
...
+                    description=description,
+                )
+                embed.set_footer(text=footer_text)
+                embeds.append(embed)
+            try:
+                await interaction.response.send_message(embeds=embeds, ephemeral=True)
+            except Exception:
+                _log.exception(f"Caught exception trying to send stats message")
+
+    # @app_commands.command(name="status", description="Display queue status")
```
Comment (StephenBartos) [inline discord_bots/cogs/common.py:366]: Please use a block comment instead haha, I hope you didn't add each # by hand
Eg:
```python
"""
Multiline 
block 
comment
"""
```

File: discord_bots/cogs/economy.py
Comment (StephenBartos) [inline discord_bots/cogs/economy.py:1]: I'm going to skip over reviewing economy, since you're way more familiar with the file and I trust your judgment here with respect to these changes

File: discord_bots/cogs/in_progress_game.py [hunk 20]
```diff
@@ -111,151 +139,281 @@ def get_player_and_in_progress_game(
...
                     ),
                     view=confirmation_buttons,
                     ephemeral=True,
                 )
                 await confirmation_buttons.wait()
                 if not confirmation_buttons.value:
                     return False
-                is_finished = await self.finish_in_progress_game(
-                    session, interaction, outcome, game_player, game
+                is_game_finished = await self.cancel_in_progress_game(
```
Comment (StephenBartos) [inline discord_bots/cogs/in_progress_game.py:190]: Good catch, since this can be confused with the internal view.is_finished field used by discord.py

File: discord_bots/cogs/in_progress_game.py [hunk 21]
```diff
@@ -111,151 +139,281 @@ def get_player_and_in_progress_game(
...
                     ),
                     view=confirmation_buttons,
                     ephemeral=True,
                 )
                 await confirmation_buttons.wait()
                 if not confirmation_buttons.value:
                     return False
-                is_game_finished = await self.cancel_in_progress_game(
-                    session, interaction, game
+                is_finished = await self.finish_in_progress_game(
```
Comment (StephenBartos) [inline discord_bots/cogs/in_progress_game.py:407]: Sorry, I got confused by the diff and thought you changed this above. Rename this to `is_game_finished`

File: discord_bots/cogs/in_progress_game.py [hunk 22]
```diff
@@ -502,14 +660,12 @@ def update_ratings(
...
         if config.GAME_HISTORY_CHANNEL:
-            game_history_channel: discord.abc.GuildChannel | None = (
-                interaction.guild.get_channel(config.GAME_HISTORY_CHANNEL)
+            game_history_channel: GuildChannel | None = interaction.guild.get_channel(
+                config.GAME_HISTORY_CHANNEL
             )
-            if game_history_channel and isinstance(
-                game_history_channel, discord.TextChannel
-            ):
+            if game_history_channel and isinstance(game_history_channel, TextChannel):
```
Comment (StephenBartos) [inline discord_bots/cogs/in_progress_game.py:668]: This can be simplified to `if isinstance(game_history_channel, TextChannel):`, since `None` is a type and will fail in the `isinstance` check

File: discord_bots/cogs/map.py [hunk 23]
```diff
@@ -1,172 +1,237 @@
...
 
-    @command()
-    @check(is_admin)
-    async def addmap(self, ctx: Context, map_full_name: str, map_short_name: str):
+    group = app_commands.Group(name="map", description="Map commands")
+
+    @group.command(name="add", description="Add a map to the map pool")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(full_name="Long name of map", short_name="Short name of map")
```
Comment (StephenBartos) [inline discord_bots/cogs/map.py:46]: Instead of outright renaming "map_short_name" to "map_name", you could have used the app_commans.rename decorator. This was probably written before you knew about it or something, but the refactor is fine.

File: discord_bots/cogs/notification.py [hunk 24]
```diff
@@ -0,0 +1,92 @@
...
+from sqlalchemy.orm.session import Session as SQLAlchemySession
+
+from discord_bots.checks import is_command_channel
+from discord_bots.cogs.base import BaseCog
+from discord_bots.models import Queue, QueueNotification, Session
+
+_log = logging.getLogger(__name__)
+
+
+class ListCommands(BaseCog):
```
Comment (StephenBartos) [inline discord_bots/cogs/notification.py:14]: This should be `class NotificationCommands`

File: discord_bots/cogs/trueskill.py [hunk 25]
```diff
@@ -0,0 +1,291 @@
...
+            )
+            if queues:
+                for queue in queues:
+                    if current in queue.name:
+                        result.append(
+                            app_commands.Choice(name=queue.name, value=queue.name)
+                        )
+        return result
+
+    ####################
```
Comment (StephenBartos) [inline discord_bots/cogs/trueskill.py:205]: block comment

File: discord_bots/utils.py [hunk 26]
```diff
@@ -62,6 +65,460 @@
 DELTA_UPPER_UNICODE = "\u03B4"
 
 
+def buildCategoryOutput(category: Category) -> str:
```
Comment (StephenBartos) [inline discord_bots/utils.py:68]: use underscore_case like the other functions. Maybe something  like `build_category_str` or `get_category_str`. Whatever you want

File: discord_bots/utils.py [hunk 27]
```diff
@@ -1016,15 +1468,21 @@ async def move_game_players_lobby(
...
     for result in results:
         # results should be empty unless an exception occured when moving a player
         if isinstance(result, BaseException):
             _log.exception("Ignored exception when moving a gameplayer to lobby:")
 
+def win_rate(wins, losses, ties):
+    denominator = max(wins + losses + ties, 1)
+    return round(
+        100 * (wins + 0.5 * ties) / denominator, 1
+    )  # why are ties counted as half a win?
```
Comment (StephenBartos) [inline discord_bots/utils.py:1484]: can remove my comment here

**Review by StephenBartos — APPROVED** [review]
Ready to merge, except for some very _very_ minor feedback. Really awesome work cleaning up all of the commands and reorganizing everything. It definitely seemed like a huge, tedious task but I really appreciate it. This is an enormous step forwards in cleaning everything up. Thank you for your patience as well!

File: discord_bots/cogs/category.py [hunk 28]
```diff
@@ -0,0 +1,337 @@
...
+    def __init__(self, bot: Bot):
+        super().__init__(bot)
+
+    group = app_commands.Group(name="category", description="Category commands")
+    
+    @group.command(name="clearqueue", description="Remove category from queue")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(queue_name="Name of existing queue")
+    async def clearqueuecategory(self, interaction: Interaction, queue_name: str):
```
[the remark this answers is not in the document]
Comment (Andrewki44) [reply discord_bots/cogs/category.py:26]: Good thinking, have moved it across to `/queue clearcategory <queue_name>`
It is a queue related command, but the intention is to remove whatever category is assigned to the given queue. 
Therefore the queue is the input, not the category.
Added it to the autocomplete as well.

File: discord_bots/cogs/category.py [hunk 29]
```diff
@@ -0,0 +1,337 @@
...
+                        colour=Colour.green(),
+                    ),
+                    ephemeral=True,
+                )
+
+    @group.command(name="create", description="Create a new category")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(name="Name of new category")
+    async def createcategory(self, interaction: Interaction, name: str):
```
[the remark this answers is not in the document]
Comment (Andrewki44) [reply discord_bots/cogs/category.py:56]: If you're looking at my dev bot, that'll be because it's synced to my `RotationConfigure` fork at the moment, which also has `CategoryConfigure` & `MapConfigure` within it.

As a part of `CategoryConfigure` this command was removed in favour of `/category configure <category_name>`.
That is not the case for this PR though, and I have tested this successful.

File: discord_bots/cogs/common.py [hunk 30]
```diff
@@ -0,0 +1,559 @@
...
+                    description="Leaderboard channel reset",
+                    colour=Colour.green(),
+                )
+            )
+
+    @app_commands.command(
+        name="stats", description="Privately displays your TrueSkill statistics"
+    )
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(category_name="Category to show stats for")
```
[the remark this answers is not in the document]
Comment (Andrewki44) [reply discord_bots/cogs/common.py:93]: Added a rename decorator.
This could be done more-so around the place, some more general clean up to be done separately.

File: discord_bots/cogs/category.py (hunk 18 above)
Comment (Andrewki44) [reply discord_bots/cogs/category.py:276]: Added a rename decorator.
Did so for the remaining functions in `category.py`

File: discord_bots/cogs/common.py (hunk 19 above)
Comment (Andrewki44) [reply discord_bots/cogs/common.py:366]: I just used the `Ctrl+/` shortcut in VS Code to be honest, and that's how it comes out.

Swapped over to docstring instead, which is convenient cause VS Code will let me collapse that now.

File: discord_bots/cogs/in_progress_game.py (hunk 21 above)
Comment (Andrewki44) [reply discord_bots/cogs/in_progress_game.py:407]: Renamed

File: discord_bots/cogs/in_progress_game.py (hunk 22 above)
Comment (Andrewki44) [reply discord_bots/cogs/in_progress_game.py:668]: Done

File: discord_bots/cogs/notification.py (hunk 24 above)
Comment (Andrewki44) [reply discord_bots/cogs/notification.py:14]: Good pickup. I was copying the class structure from file to file and must have missed the rename on this one, thanks!
Weird that it didn't give any kind of error though, I guess cause on sync it's syncing the command tree via the unique group.

File: discord_bots/cogs/trueskill.py (hunk 25 above)
Comment (Andrewki44) [reply discord_bots/cogs/trueskill.py:205]: Done

File: discord_bots/utils.py (hunk 26 above)
Comment (Andrewki44) [reply discord_bots/utils.py:68]: I believe this was added in a PR after this was originally raised, so was likely missed from my review.
Good catch, have renamed it here, and in all references.

File: discord_bots/utils.py (hunk 27 above)
Comment (Andrewki44) [reply discord_bots/utils.py:1484]: Done

**Comment by Andrewki44** [comment]
@StephenBartos all comments have been reviewed and updated. 
Should be ready to merge.
